### PR TITLE
Add authentication APIs for admin and students.

### PR DESCRIPTION
**Pull request by abhinavsingh06**

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
+var express = require("express");
 var router = express.Router();
+const Auth = require("../utils/auth");
+const User = require("../models/User");
 
-/* GET home page. */
-router.get('/', function(req, res, next) {
-  res.render('index', { title: 'Express' });
+//get dashboard
+router.get("/dashboard", Auth.verToken, (req, res) => {
```
Comment (prank7) [inline routes/index.js]: Naming is not right. What is Auth.verToken?

It should be `Auth.verifyToken`. No shortcuts.

File: routes/index.js [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+  if (req.isAdmin)
```
Comment (prank7) [inline routes/index.js]: Put the if statement in curly braces explicitly.

File: utils/auth.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const jwt = require("jsonwebtoken");
+const User = require("../models/User");
+
+exports.genToken = userId => {
```
Comment (prank7) [inline utils/auth.js]: Name it `generateToken`

**Review by prank7 — CHANGES_REQUESTED** [review]

File: routes/index.js (hunk 1 above)
Comment (abhinavsingh06) [reply routes/index.js]: Resolved

File: routes/index.js (hunk 2 above)
Comment (abhinavsingh06) [reply routes/index.js]: Resolved

File: utils/auth.js (hunk 3 above)
Comment (abhinavsingh06) [reply utils/auth.js]: Resolved

File: utils/auth.js [hunk 4]
```diff
@@ -1,11 +1,11 @@
 const jwt = require("jsonwebtoken");
 const User = require("../models/User");
 
-exports.genToken = userId => {
+exports.generateToken = userId => {
```
Comment (prank7) [inline utils/auth.js]: `module.exports = {
  fn1, 
  fn2
}`

**Review by prank7 — CHANGES_REQUESTED** [review]

File: utils/auth.js (hunk 4 above)
Comment (abhinavsingh06) [reply utils/auth.js]: Resolved

**Pull request merged**